### ISSUE TITLE
allow disabling cert checks

### ISSUE THREAD

**Issue by brancz** [coreos/prometheus-operator#70]
The Prometheus TLS config has the possibility to allow skipping certificate verification. We allow setting the `scheme`, so we also need to be able to set whether to verify the certificates.

This blocks coreos/kube-prometheus#21 as there are on going certificate issues with the kubelet and apiserver.

@fabxc @alexsomesan

**Comment by brancz** [comment]
I propose a boolean in the [`Endpoint`](https://github.com/coreos/prometheus-operator/blob/854bab6774af287000a9be2f5ca20f94c9ebe3cd/Documentation/service-monitor.md#endpoint) named the same as in Prometheus: `insecure_skip_verify`. Let me know what you think @fabxc @alexsomesan .

**Comment by brancz** [comment]
Unfortunately we need more than that basically we need to be able to specify all the following:

```
      tls_config:
        ca_file: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
        # Skip verification until we have resolved why the certificate validation
        # for the kubelet on API server nodes fail.
        insecure_skip_verify: true
      bearer_token_file: /var/run/secrets/kubernetes.io/serviceaccount/token
```

However, that feels like we're recreating the Prometheus config now, wdyt @fabxc @alexsomesan ? Technically it's "only" the tls and auth section (relabelling is still automated).

**Comment by brancz** [comment]
@fabxc do you think we should generally allow users to specify these things or simply based on a flag in the Endpoint object of the ServiceMonitor set these things based on "isKubernetesComponent: (true|false)"?

**Comment by fabxc** [comment]
There's no real way around configuring auth all auth params. 
Users have no way of using bearer token files though AFAICS, as we don't let them mount configmaps into their pods. Worth thinking about the options and how to make them user friendly.

Certainly no snake case though.

**Comment by gianrubio** [comment]
@brancz I arrived here because I'm getting an error when scrapping my apiserver trough my service auto discovery and now there's no way to skip the ssl validation.

```
Get https://10.10.128.1:443/metrics: x509: certificate is valid for 10.10.12.12, not 10.10.128.1
```

I propose is to specify the ```insecure_skip_verify```  using an annotation similar this

```
apiVersion: v1
kind: Service
metadata:
  name: kube-apiserver-prometheus-discovery
  labels:
    k8s-app: kubernetes
  annotations:
    annotations: prometheus.io/insecure_skip_verify: true
spec:
  selector:
    k8s-app: kube-apiserver
  type: ClusterIP
  clusterIP: None
  ports:
  - name: https-metrics
    port: 443
    targetPort: 443
    protocol: TCP
```

**Comment by brancz** [comment]
That would probably have to be a change in prometheus upstream. Best would of course be if we generate the certs correctly, but there is a similar problem with the kubelet, where certs are generated on the fly if none are given, which means they are self signed, instead of signed with the ca for your cluster. Therefore we definitely need a solution.

**Comment by brancz** [comment]
Closed with #103
